src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post.
+ *     responses:
+ *       200:
+ *         description: The top three likes for the post.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 likes:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *                     properties:
+ *                       _id:
+ *                         type: string
+ *                         description: The unique identifier of the like.
+ *                       user_id:
+ *                         type: string
+ *                         description: The ID of the user who liked the post.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ */
 router.get('/topThreeLikes/:postID', getTopThreeLikes);
+
+/**
+ * @swagger
+ * /posts/countMyPosts:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Count the number of posts created by the current user , Must be connected as user
+ *     description: Retrieves the total number of posts created by the user identified by the token in the request. Ensure that the request includes authentication middleware to set `req.tokenData`.
+ *     security:
+ *       - apiKeyAuth: []   # This line adds the x-api-key to this specific route
+ *     responses:
+ *       200:
+ *         description: The number of posts created by the user.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 count:
+ *                   type: integer
+ *                   description: The total number of posts created by the user.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message.
+ *                 err:
+ *                   type: string
+ *                   description: Details of the error.
+ *     components:
+ *       securitySchemes:
+ *         BearerAuth:
+ *           type: http
+ *           scheme: bearer
+ *           bearerFormat: JWT
+ */
 router.get('/countMyPosts', auth, countMyPosts);
+
+/**
+ * @swagger
+ * /users/{userID}/posts:
+ *   get:
+ *     tags:
+ *       - Posts operations
+ *     summary: Retrieve posts created by a specific user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"